WIP: Replace docstring bubble with right sidebar preview

rabea-al opened the pull request:
# Description

- Replaced the old chat-bubble style that appeared when clicking the ℹ︎ icon on a component with a new preview panel that opens in the right sidebar.
- Updated the infoIcon to support both light and dark themes automatically.


## Pull Request Type

- [ ] Xircuits Core (Jupyterlab Related changes)
- [ ] Xircuits Canvas (Custom RD Related changes)
- [ ] Xircuits Component Library
- [ ] Xircuits Project Template
- [ ] Testing Automation
- [ ] Documentation
- [ ] Others (Please Specify)

## Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Tests

1. Launch Binder (or run locally).
2. Drag any component into the canvas.
3. Click the **ℹ︎** icon on the node.

   * The right sidebar should expand and show the component name and docstring.
4. Click the same ℹ︎ again → the sidebar collapses.
5. Click ℹ︎ on a different component → the sidebar swaps to the new docstring.


**Tested on? Specify Version.**

- [ ] Windows  
- [x] Linux Fedora
- [ ] Mac  
- [ ] Others  (State here -> xxx )

rabea-al commented:
What’s new in this PR:
- Added navigation arrows to move between directly connected nodes (excluding Start and Finish).
- Automatically centers the selected node when navigating.
- Added a button to open the script of the selected component node.
- Added a button to center the current node manually.
- Added a special button to open the full workflow (.xircuits) if the current node is of type xircuits_workflow.
- This button is conditionally shown only when the node is of that type.

MFA-X-AI reviewed (commented):
Thanks for the PR! This is a big feature, so here's a longer review:

- The sidebar preview has quite a large minimum sized window. I think it should follow the minimum width as the other sidebars.

https://github.com/user-attachments/assets/63dde143-19af-46d7-ab67-67cf5167156b

- The previous and next node logic shouldn't go to Literals and Arguments. 

https://github.com/user-attachments/assets/3133913d-2d5c-4190-9767-5550bbe5ff0d

I think you should allow it to go to Start and Finish and display a simple message like, "This is the start or end of your workflow!" 

- The component preview doesn't switch context when another canvas is active. I think it should just reset once a new workflow is opened. 

https://github.com/user-attachments/assets/ed72d959-3af2-4c0d-82a3-46d999244ecd

- This does mean it might be complex if there is 2 active workflows for whatever reason. If that's the case, you might need to display on the component preview which component is this workflow coming from. I think that's a good thing to have anyways. Maybe under the Component preview header.

- I think we can improve the current preview for blank canvas. I'll pass you a placeholder image. 

<img width="561" height="809" alt="image" src="https://github.com/user-attachments/assets/8748ed1f-9f4c-4981-aebf-55553d0062aa" />

- workflow components doesn't seem to have an `i` button yet. I guess that's something we haven't added. 
<img width="310" height="156" alt="image" src="https://github.com/user-attachments/assets/d06ad3cc-11c0-4158-aaba-d872c421c540" />

Since this is a info docstring PR, could you add it in?

- The handling for the previous and next node for branch component might be a bit more complex. I think the best way is go to the top paths only. 

https://github.com/user-attachments/assets/69f27035-00f4-4602-ab6b-0a8e31566c47

okay, that's some things to do for now. Thanks a bunch.

rabea-al commented:
**Final Changes Implemented Based on Review Feedback:**

 **Sidebar Width**

* Adjusted the sidebar preview to follow the minimum width like other sidebars for consistent layout.

 **Previous/Next Navigation**

* Skipped navigation to `Literal` and `Argument` nodes.
* Allowed navigation to `Start` and `Finish` nodes, with message:

  * *“This is the start of your workflow.”*
  * *“This is the end of your workflow.”*
* Handled navigation logic in branch components by only traversing the **top path**, as suggested.

 **Canvas Switching**

* When switching to a different canvas, the component preview now **resets and collapses** automatically to prevent confusion.

 **Workflow Components**

* Added the `ℹ` (info) toggle button to **workflow components**, enabling preview for sub-workflows.
* If the workflow is a Python file, clicking "Open workflow" now tries to open the `.xircuits` version.

MFA-X-AI reviewed (approved):
Working very well! I've pushed a string update for the empty panel preview, since I realized we do need the users to click on the `i` icon. 
I'll merge it, thanks for the PR!
